### PR TITLE
Fix progress bar showing finished status for unselected reals

### LINKED ISSUE #9230: Progress bar shows finished status for non existing realizations
### What happened? (You can include a screenshot if it helps explain)

Started only 3 realizations on/of an ensemble experiment. Progress bar shows 247 finished realizations which does not exist. 
Image

### What did you expect to happen?

Progress bar and status only to include the 3 realizations.

### steps to reproduce

```Python

```

### Environment where bug has been observed

- [ ] python 3.10
- [ ] python 3.11
- [ ] python 3.12
- [ ] macosx
- [ ] rhel7
- [x] rhel8
- [ ] local queue
- [x] lsf queue
- [ ] slurm queue
- [ ] openPBS queue

### PR DESCRIPTION
**Pull request by jonathan-eq**
**Issue**
Resolves #9230

**Approach**
This commit fixes the bug where the gui shows the total count of realizations as the ensemble_size rather than the number of realizations actually selected for the run.

(Screenshot of new behavior in GUI if applicable)
image

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'pytest tests/ert/unit_tests -n logical -m "not integration_test"'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/run_models/base_run_model.py [hunk 1]
```diff
@@ -406,7 +405,9 @@ def get_current_status(self) -> dict[str, int]:
                 for real in all_realizations.values():
                     status[str(real["status"])] += 1
 
-        status["Finished"] += self.active_realizations.count(False)
+        status["Finished"] += self.active_realizations.count(
```
Comment (xjules) [inline src/ert/run_models/base_run_model.py:408]: It would be nice to have a small comment / explanation here.

Comment (jonathan-eq) [reply src/ert/run_models/base_run_model.py:408]: Yes, either a comment or put it in a separate function and have a unit test on it.
The problem was that when we added all the `self.active_realizations.count(False)`, we also add the realizations that have deselected from the run. 
The idea was to use the the original `'active_realizations'`-list (which was already an instance variable `self_initial_realizations_mask`), and subtract this from the previous number.

File: src/ert/run_models/base_run_model.py [hunk 2]
```diff
@@ -406,9 +405,14 @@ def get_current_status(self) -> dict[str, int]:
                 for real in all_realizations.values():
                     status[str(real["status"])] += 1
 
-        status["Finished"] += self.active_realizations.count(False)
+        status["Finished"] += self._get_number_of_finished_realizations_from_reruns()
         return status
 
+    def _get_number_of_finished_realizations_from_reruns(self) -> int:
```
Comment (xjules) [inline src/ert/run_models/base_run_model.py:411]: Nice!

**Review by xjules — APPROVED** [review]
Nice PR @jonathan-eq ! :rocket:

**Pull request merged**